OpenRA.Mods.Common/AI/HackyAI.cs

```diff
@@ -984,7 +987,11 @@ void BuildUnit(string category, bool buildRandom)
 				ChooseRandomUnitToBuild(queue) :
 				ChooseUnitToBuild(queue);
 
-			if (unit != null && Info.UnitsToBuild.Any(u => u.Key == unit.Name))
+			if (unit != null
+				&& Info.UnitsToBuild.ContainsKey(unit.Name)
+				&& (!Info.UnitLimits.ContainsKey(unit.Name)
```

Comment:
This needs a null check.

https://github.com/Phrohdoh/OpenRA/issues/15#issuecomment-141710307